extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/PythonTableDataService.java

```diff
@@ -343,35 +394,37 @@ public List<WritableChunk<Values>> getColumnValues(
                 final int minimumSize,
                 final int maximumSize) {
 
-            final ArrayList<WritableChunk<Values>> resultChunks = new ArrayList<>();
+            final AsyncState<List<WritableChunk<Values>>> asyncState = new AsyncState<>();
+
             final String columnName = columnDefinition.getName();
             final Consumer<ByteBuffer[]> onMessages = messages -> {
                 if (messages.length < 2) {
-                    throw new IllegalArgumentException(String.format("%s:%s: column_values(%s, %d, %d, %d) expected at "
-                            + "least 2 IPC messages describing the wire format of the column followed by column "
-                            + "values, but received %d messages", tableKey, tableLocationKey, columnName,
-                            firstRowPosition, minimumSize, maximumSize, messages.length));
+                    asyncState.setError(new IllegalArgumentException(String.format(
+                            "expected at least 2 IPC messages describing the wire format of the column followed by "
+                                    + "column values, but received %d messages",
+                            messages.length)));
+                    return;
                 }
-                resultChunks.ensureCapacity(messages.length - 1);
+                final ArrayList<WritableChunk<Values>> resultChunks = new ArrayList<>(messages.length - 1);
 
                 final Schema schema = ArrowToTableConverter.parseArrowSchema(
                         ArrowToTableConverter.parseArrowIpcMessage(messages[0]));
                 final BarrageUtil.ConvertedArrowSchema schemaPlus = BarrageUtil.convertArrowSchema(schema);
 
                 if (schema.fieldsLength() > 1) {
-                    throw new IllegalArgumentException(String.format("%s:%s: column_values(%s, %d, %d, %d) received "
-                            + "more than one field. Received %d fields for columns %s",
-                            tableKey, tableLocationKey, columnName, firstRowPosition, minimumSize, maximumSize,
+                    asyncState.setError(new IllegalArgumentException(String.format(
```

Comment:
I think you need/want to throw in these cases, and let the catch block handle cleanup. As-is, you are leaking chunks.